test/e2e/e2e/utils.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2020 Cortex Labs, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+from http import HTTPStatus
+from typing import List, Optional, Dict, Union, Callable
+
+import cortex as cx
+import requests
+import yaml
+
+
+def wait_for(fn: Callable[[], bool], timeout=None) -> bool:
+    deadline = time.time() + timeout if timeout else None
+    while True:
+        if deadline is not None and time.time() > deadline:
+            return False
+
+        done = fn()
+        if done:
+            return True
+
+        time.sleep(1)
+
+
+def apis_ready(client: cx.Client, api_names: List[str], timeout: Optional[int] = None) -> bool:
+    def _is_ready():
+        return all(
+            [client.get_api(name)["status"]["status_code"] == "status_live" for name in api_names]
+        )
+
+    return wait_for(_is_ready, timeout=timeout)
+
+
+def endpoint_ready(client: cx.Client, api_name: str, timeout: int = None) -> bool:
+    def _is_ready():
+        endpoint = client.get_api(api_name)["endpoint"]
+        response = requests.post(endpoint)
+        return response.status_code == HTTPStatus.BAD_REQUEST
```

Comment:
I believe the APIs will respond with a 200 if you make a GET to the API endpoint, in case you want to consider using that here

Reply:
couldn't get that to work